src/libs/sys-util-rs/src/hooks.rs

```diff
@@ -0,0 +1,423 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::hash::{Hash, Hasher};
+use std::io::{self, Read, Result};
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use lazy_static::lazy_static;
+use subprocess::{ExitStatus, Popen, PopenConfig, PopenError, Redirection};
+
+use crate::{eother, sl};
+
+const DEFAULT_HOOK_TIMEOUT: i32 = 10;
+
+/// Execution state of OCI hooks.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum HookState {
+    /// Hook is pending for executing/retry.
+    Pending,
+    /// Hook has been successfully executed.
+    Done,
+    /// Hook has been marked as ignore.
+    Ignored,
+}
+
+lazy_static! {
+    static ref EXEC_RESULT: Arc<Mutex<HashMap<u64, HookState>>> =
+        Arc::new(Mutex::new(HashMap::new()));
```

Comment:
Since this is a unique global, I don't think you need the `Arc`, just the `Mutex`.  Although, TBH, I'd probably prefer an explicit `HookContext` or whatever argument, rather than using a global.